src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonMigrationService.kt

```diff
@@ -59,47 +62,86 @@ class PrisonPersonMigrationService(
   override suspend fun migrateNomisEntity(context: MigrationContext<PrisonerId>) {
     log.info("attempting to migrate ${context.body}")
     val offenderNo = context.body.offenderNo
-    val telemetry = mutableMapOf("offenderNo" to offenderNo)
+    val telemetry = mutableMapOf(
+      "offenderNo" to offenderNo,
+      "migrationId" to context.migrationId,
+      "migrationType" to PHYSICAL_ATTRIBUTES,
+    )
 
-    // TODO wrap this in a try catch for negative telemetry
-    val physicalAttributes = prisonPersonNomisApiService.getPhysicalAttributes(offenderNo)
-    val dpsIds = physicalAttributes.bookings.flatMap { booking ->
-      booking.physicalAttributes.map { pa ->
-        val (lastModifiedAt, lastModifiedBy) = pa.lastModified()
-        prisonPersonDpsService.migratePhysicalAttributesRequest(
-          heightCentimetres = pa.heightCentimetres,
-          weightKilograms = pa.weightKilograms,
-          appliesFrom = booking.startDateTime.toLocalDateTime(),
-          appliesTo = booking.endDateTime?.toLocalDateTime(),
-          createdAt = lastModifiedAt,
-          createdBy = lastModifiedBy,
-        )
-      }
-    }.let { requests ->
-      prisonPersonDpsService.migratePhysicalAttributes(offenderNo, requests)
-    }.fieldHistoryInserted
-    telemetry["dpsIds"] = dpsIds.toString()
+    try {
+      val dpsIds = prisonPersonNomisApiService.getPhysicalAttributes(offenderNo)
+        .toDpsMigrationRequest()
+        .migrate(offenderNo)
+      telemetry["dpsIds"] = dpsIds.toString()
 
-    prisonPersonMappingService.createMapping(
       PrisonPersonMigrationMappingRequest(
         nomisPrisonerNumber = offenderNo,
         migrationType = PHYSICAL_ATTRIBUTES,
         label = context.migrationId,
         dpsIds = dpsIds,
-      ),
-      object : ParameterizedTypeReference<DuplicateErrorResponse<PrisonPersonMigrationMappingRequest>>() {},
-    )
+      ).createActivityMapping(context)
 
-    telemetryClient.trackEvent(
-      "prisonperson-migration-entity-migrated",
-      mapOf(
-        "offenderNo" to offenderNo,
-        "migrationId" to context.migrationId,
-        "dpsIds" to dpsIds,
-      ),
-    )
+      telemetryClient.trackEvent("prisonperson-migration-entity-migrated", telemetry)
+    } catch (e: Exception) {
+      telemetry["error"] = e.message ?: "unknown error"
+      telemetryClient.trackEvent("prisonperson-migration-entity-failed", telemetry)
+      throw e
+    }
+  }
```

Comment:
This is the main change - add exception handling and tidy up the code for readability